src/cryptonote_core/tx_pool.cpp

```diff
@@ -131,7 +131,7 @@ namespace cryptonote
 
     // we do not accept transactions that timed out before, unless they're
     // kept_by_block
-    if (!kept_by_block && m_timed_out_transactions.find(id) != m_timed_out_transactions.end())
+    if (tx_relay != relay_method::block && m_timed_out_transactions.find(id) != m_timed_out_transactions.end())
```

Comment:
Could have had a const bool kepy_by_block = tx_relay == relay_method::block above to save all the diffs below.

Reply:
Ok, will do.